FWCore/ParameterSet/test/test_catch_ParameterSetDescription.cc

```diff
@@ -1643,6 +1643,136 @@ p = dict(
     psetDesc.validate(params);
   }
 
+  SECTION("obsolete") {
+    SECTION("string") {
+      edm::ParameterSetDescription psetDesc;
+
+      psetDesc.add<std::string>("testname");
+      psetDesc.add<std::string>("hadDefault", "default");
+      psetDesc.addObsolete<std::string>("noLongerUsed");
+      SECTION("with obsolete") {
+        edm::ParameterSet params;
+        params.addParameter<std::string>("testname", std::string("testvalue"));
+        params.addParameter<std::string>("noLongerUsed", std::string("testvalue"));
+
+        psetDesc.validate(params);
+      }
+      SECTION("without obsolete") {
+        edm::ParameterSet params;
+        params.addParameter<std::string>("testname", std::string("testvalue"));
+        psetDesc.validate(params);
+      }
```

Comment:
How about asserting there the `noLongerUsed` parameter was not added?

Reply:
done